src/modules/Tasks.js

```diff
@@ -0,0 +1,52 @@
+class Tasks {
+  constructor() {
+    this.tasks = JSON.parse(window.localStorage.getItem('todo.tasks') || '[]');
+  }
+
+  addTask(task) {
+    this.tasks.push(task);
+
+    this.updateLocalStorage();
+  }
+
+  getSortedTasks() {
+    return this.tasks.sort((a, b) => a.index - b.index);
+  }
+
+  removeTask(task) {
+    const index = this.tasks.findIndex((t) => task.index === t.index);
+
+    if (index === -1) return;
+
+    this.tasks.splice(index, 1);
+
+    this.tasks = this.tasks.map((task, index) => ({
+      ...task,
+      index,
+    }));
+
+    this.updateLocalStorage();
+  }
+
+  editTask(task, newTask) {
+    const index = this.tasks.findIndex((t) => task.index === t.index);
+
+    if (index === -1) return;
+
+    this.tasks[index] = newTask;
+
+    this.updateLocalStorage();
+  }
+
+  updateLocalStorage() {
+    window.localStorage.setItem('todo.tasks', JSON.stringify(this.tasks));
+  }
+
+  clearCompletedTasks() {
+    this.tasks = this.tasks.filter((task) => !task.completed);
+
+    this.updateLocalStorage();
+  }
```

Comment:
- [ ] Great job so far :+1: The completed key changes when a user clicks the checkbox and all completed tasks are deleted when the `clear all completed` button is clicked :+1: Kindly update the indexes of the remaining elements after clearing the completed tasks :+1: 
- [ ] (OPTIONAL) Your index should begin from 1 and not 0. It would be great if you could rectify this issue :100: